canonical/react-components
# feat(skip-link): add SkipLink component [WD-15080]

## Done

- Added new `SkipLink` component, based on Vanilla [`p-skip--link`](https://vanillaframework.io/docs/patterns/links#skip-link) pattern. Added tests and Storybook for it.
- Integrated the `SkipLink` component inside the `ApplicationLayout`, so that it can be automatically obtained by React applications using the `ApplicationLayout`.
- :warning: **If you are manually adding an `a` element with the `p-skip--link` class to your React application, and are also using the `ApplicationLayout` component, this can be a breaking change for you**.
Please make sure that you remove your manual addition of the skip link when upgrading react-components to the version that includes this change.

## QA

### Storybook

To see rendered examples of all react-components, run:

```shell
yarn start
```

### QA in your project

from `react-components` run:

```shell
yarn build
npm pack
```

Install the resulting tarball in your project with:

```shell
yarn add <path-to-tarball>
```

### QA steps

- Go to the `SkipLink` page of Storybook.
- Click inside the container of the default example, to move the focus there.
- Hit "Tab", and check that the "Skip to main content" link appears.

### Percy steps

- There should be a new page in Storybook, for the `SkipLink` component.

## Fixes

Fixes: [WD-15080](https://warthogs.atlassian.net/browse/WD-15080)

[WD-15080]: https://warthogs.atlassian.net/browse/WD-15080?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
